Extend insights api

shschwar opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [V ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet. Nov 2020
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet. Nov 2020
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [X ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

Commit:
Add queries API and getInsightsAPI

Add queries and getInsights without examples

Commit:
Add queries API and getInsightsAPI

Add queries and getInsights without examples

Commit:
Update file

Commit:
Update file

Commit:
Add Examples

2 example files, GetQuery,PosttGetInsights

Commit:
Add Examples

2 example files, GetQuery,PosttGetInsights

riserrad committed:
Fix Swagger Linting issues reported (#11462)

* Fixing linting issues according to report (task 8554751)

* Adding a default response to operations

jimmyca15 committed:
Amend error response to indicate that we return additional error info. (#11678)

asmaskar committed:
Adding Identity to Patch Vault (#11646)

Xynoclafe committed:
Create 2020-10-01 version of resources swagger with templateLink.queryString property (#11637)

* Initial commit

* Add templateLink.queryString property into 2020-10 resources version

* Prettier fix

* Fix api-version in examples

* Change description of queryString

* Fixed an undefined tag in readme

* Avocado fixes

* Suppress pre-existing lint errors

* Fix template specs tag

* Revert "Fix template specs tag"

This reverts commit c0a2358b1ff17f1a6e278a79e9a691b4226f1ac4.

* For real this time: Fix template specs tag

Co-authored-by: Filiz Topatan <fitopata@microsoft.com>

sagamzu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -10539,7 +10644,187 @@
+    "EntityGetInsightsParameters": {
+      "description": "The parameters required to execute insights operation on the given entity.",
+      "properties": {        
+        "queryKind": {
```
why need queryKind if the API is only for insights?

sagamzu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -10539,7 +10644,187 @@
+          "description": "Indicates if query time range should be extended with default time range of the query. Default value is false",
+          "type": "boolean"
+        },
+        "insightQueryId":{
```
maybe change insightQueryId to insightQueryId*s*

sagamzu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -10539,7 +10644,187 @@
+        "queryTimeInterval": {
+          "type": "object",
+          "description": "The Time interval that the query actually executed on.",
+              "properties": {
```
some indentation not looks good. I am not sure if it is only on my computer..
did you execute prettier?

sagamzu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -10539,7 +10644,187 @@
+        },
+        "tableQueryResults": {
+          "$ref": "#/definitions/InsightsTableResult",
+          "description": Query results for table insights query."
```
Missing " before Query. 

Please reun the scripts to test the file & example

sagamzu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -10734,6 +11019,30 @@
+      "name": "kind",
+      "required": true,
+      "type": "string",
+      enum: [
```
missing "" for enum

sagamzu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -2191,6 +2191,111 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/entities/{entityId}/queries": {
```
can you please remined me why we need this API?

sagamzu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -10734,6 +11019,30 @@
+      },
+      "x-ms-parameter-location": "method"
+    },
+    "InsightActivityKind": {
```
Mayb erename `InsightActivityKind`? It has activity & insight.

sagamzu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -2191,6 +2191,111 @@
+            "schema": {
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/EntityInsightItem"
```
Is it only for insight or it will return also activities? because In `InsightActivityKind `there is both activity & insughts

shschwar replied:
Both

sagamzu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -10539,7 +10644,187 @@
+        "kind": {
+          "type": "string",
+          "description": "Insight or Activity type",
+          "enum": [
```
I am not sure about the way we need to define enum as string. 
In some places I see diferent defenition, that include` x-ms-enum`

for example:
```
"enum": [
            "Scheduled",
            "MicrosoftSecurityIncidentCreation",
            "Fusion"
          ],
          "type": "string",
          "x-ms-enum": {
            "modelAsString": true,
            "name": "AlertRuleKind",
            "values": [
              {
                "value": "Scheduled"
              },
              {
                "value": "MicrosoftSecurityIncidentCreation"
              },
              {
                "value": "Fusion"
              }
            ]
          }
```

shschwar replied:
[interesting point ](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-enum).
It seems that x-ms-enum are not enum, but string with documented recommended value, while enum is stricter.
It seems that using enum should be for API that rarely change, while x-ms-enum should be for list of values that often changed.

lordjeb committed:
Add WVD Scaling Plan resource definitions (#11224)

Co-authored-by: Jeremy Hurren <jehurren@microsoft.com>

cataggar committed:
fix AVS 2020-03-20 correctness errors for empty body and internet default (#11486)

* backport empty body and internet default fixes

* leave flatten on

changlong-liu committed:
hide StorageAccountCredentials put (#11670)

* hide StorageAccountCredentials put

* hide StorageAccounts put

adhishchugh committed:
add traceLevel, continueOnError, and runConcurrently properties to Dataflow activity properties (#11640)

Commit:
Fix custom-words.txt (#11696)

idear1203 committed:
[Synapse] - Fix typo in SqlServer Swagger spec (#11567)

Co-authored-by: Dongwei Wang <dongwwa@microsoft.com>

tunafish88 committed:
Removing encryptionprotector APIs as this can land the service in an incosttent state (#11663)

arolshan committed:
Update readme.md (#11564)

dulikvor committed:
Microsoft.OperationalInsights: Updating property name + Adding a new resource (#11649)

* Updating property name.

* example update.

* Adding force cmk to workspace.

* Adding default error response.

* Removing redundant examples.

* adding workspace capping to example per code review request.

* Per review request.

Co-authored-by: Dudi Likvornik <dulikvor@microsoft.com>

ArcturusZhang committed:
Add new api-version for databoxedge (#11694)

sagamzu commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -2191,6 +2191,111 @@
+            "$ref": "#/parameters/EntityId"
+          },
+              {
+            "$ref": "#/parameters/InsightActivityKind"
```
rename to general name

greenms committed:
[Hub Generated] Review request for Microsoft.Cache to add version preview/2020-10-01-preview (#11683)

* New Readme Config File

* New Go Language Readme Config File

* New Typescript Language Readme Config File

* New Python Language Readme Config File

* New C# Language Readme Config File

* New Ruby Language Readme Config File

* New Swagger Spec File

* New Swagger Example Spec File

* Initial updates for public preview

* Add default error response. Add integer format.

* Add 200 response to delete operations.

* Add custom words for Redis Enterprise.

* Update whitespace for prettier.

* Update whitespace for prettier.

* Update whitespace for prettier.

* Revert "Update whitespace for prettier."

This reverts commit 4f04b1bfbf35407c783a3249e783ba4b34590ea9.

* Revert "Update whitespace for prettier."

This reverts commit ccca6d5e636f2e3aa38f80c735638e8445a6c42e.

* Revert "Update whitespace for prettier."

This reverts commit 3d72e6bed67117d7a407ba5fd467673f0f12d16b.

* Update whitespace for prettier.

* Update Go settings

* Including .NET SDK in list of SDKs automatically generated.

* Update csharp settings.

* Fix typo.

* Updates addressing comments from reviewer.

* Fix provisioningState errors.

* Update clusteringPolicy description and related examples.

* Remove Enterprise_E1 SKU.

* Updates per comments from reviewer.

* Remove illegal terminal states from provisioningState. Add new resourceState.

* Utilize common properties from updated v2 types.json

* Add 'Deleting' state to ProvisioningState enum.

* Make operationIds of Database operations consistent.

Commit:
IoTC 2011 Adding jobs, DTDLv2 templates, groups (#11622)

* Adding jobs, DTDLv2 templates, groups

* Spelling mistakes fix

* Trying to suppress email lint

* Small update to fix linter

* Added additional suppression for long running ops

* Again

* Updated again

* Removing the other suppressors

* New suppression without quotes

* back to just email suppression

Commit:
Execute validation tools and fic results

Commit:
Execute validation tools and fic results

Commit:
Fix according to comments

Commit:
Fix according to comments

rapatank committed:
Update to use common type (#11710)

ChenTanyi committed:
fix java readme (#11695)

amoghnatu committed:
Fixed some more Swagger issues (#11713)

Co-authored-by: Amogh Natu <amnat@microsoft.com>

pilor committed:
Microsoft.PolicyInsights/checkPolicyRestrictions API (#10798)

* CheckPolicyRestrictions API

* Run prettier

* Increment package version

* Remove from old tag

* Remove suppression

zhiyong-gayang committed:
Change PartitionOption to object to allow expression (#11613)

* add httpRequestTimeout to odata based connector activities

* add httpRequestTimeout to odata based connector activities

* add sql parallel read properties

* change partitionOption to object

* change ParallelOption to object

herreraj-ms committed:
[Monitor] Add DataCollRules section  readme.go.md (#11707)

changlong-liu committed:
20201113 streamanalytics (#11671)

* sdk-no-flatten

* use unique jobName is examples

vladrai committed:
s360 fixes for Azure DNS (#11591)

* s360 fixes for Azure DNS

* Additional property

* Property fix

* Readonly property

* CloudError fix

* Examples. cloud error

* Avocado fix

* read.md

* Prettier

hunngu-ms committed:
Threat Intelligence: Update name and descriptions of definitions, parameters (#10949)

* Update API descriptions

* Resolve Comments

* Update parameter name of create/update indicator apis

* Fix example issue

mit2nil committed:
Fix support swagger correctness violation (#11709)

* updated version, example and schema changes

* fixing examples and updating readme files

* Update support.json by gangan

Removed html tags (kept some <br/> but can remove them as well)
moved all of the links to md style.

* Update support.json

Added CSP and Sev 1 description

* Updates from swagger json review

* Updates from examples review

* prettier check

* updating custom words and running prettier

* fixing path in readme.go.md

* Reverted change from invalid upstream merfe

* Fix swagger correctness violation for integer type format

* Try to change format to int64 since int32 is flagged as breaking change

* Revert "Try to change format to int64 since int32 is flagged as breaking change"

This reverts commit cb602d9fcec11dee2a5bed30d86f4d329f74d47b.

Co-authored-by: Bhavin Shah <bhshah@microsoft.com>
Co-authored-by: shahbj79 <shahbj79@hotmail.com>
Co-authored-by: Ganga Narayanan <gangan@microsoft.com>
Co-authored-by: Nilay Chheda <nichheda@microsoft.com>

Commit:
Add abstract

Commit:
Add abstract

srkakuru committed:
Add new synapse rbac api spec (#11384)

* Add files via upload

* Fix validation issues

* Fix validation issues

* Fix prettier check issue

* fix validation errors

* Made review fixes

* Run prettier check

* Review fix

Co-authored-by: Dongwei Wang <dongwwa@microsoft.com>

Commit:
Execute auto correction

Commit:
Execute auto correction

Commit:
Split ActivityTimelineItem

Commit:
Split ActivityTimelineItem

dhawal777 committed:
Specifying shipping address required property to autorest. (#11717)

* Ensuring autorest generation follow specific order of required properties in shippingAddress

* added details  sharp yaml

tunafish88 committed:
Removing renameSqlPool api (#11664)

Commit:
[ACR] SystemData and new readonly KeyVaultProperties (#11601)

harshivcodes committed:
[Hub Generated] Review request for Microsoft.StoragePool to add version preview/2020-03-15-preview (#11718)

* New Readme Config File

* New Go Language Readme Config File

* New Typescript Language Readme Config File

* New Python Language Readme Config File

* New C# Language Readme Config File

* New Ruby Language Readme Config File

* New Swagger Spec File

* New Swagger Example Spec File

* StoragePool RP swagger definition

* Added missing file

* Added missing title and description

* Prettier fix

* Fixing js code gen issue

Co-authored-by: Harshitha Akkaraju <hakkaraj@microsoft.com>

yangyuan committed:
Swagger Correctness for Cognitive Services (#11405)

* Swagger Correctness

* Undo changes in 2016-02-01-preview

* Move location of field

* change enum type to string for Sku.tier

juhee0202 committed:
Microsoft.Capacity 2020-10-01-preview update (#11668)

jianyexi committed:
Update openapi-authoring-automated-guidelines.md to avoid comfusing (#11208)

* Update openapi-authoring-automated-guidelines.md

* Update openapi-authoring-automated-guidelines.md

* change url of suppressing process

harics24 committed:
Reverting the changed made for the address space update in peered vNets. PR: 11090 (#11731)

Co-authored-by: Hari Prasad Perabattula <haperaba@microsoft.com>

zhejun97 committed:
set readonly property of contentStorageAccount to false (#11630)

Co-authored-by: Zhejun Zhao <t-zhzhao@microsoft.com>

Commit:
Split ActivityTimelineItem into 2 different objects

Commit:
Split ActivityTimelineItem into 2 different objects

idear1203 committed:
Add Nullable annotation (#11684)

Co-authored-by: Dongwei Wang <dongwwa@microsoft.com>

xolve committed:
Revert missing response code. (#11690)

* Revert missing response code.

* Prettier format fix.

Co-authored-by: Ravi Arora <raviarora@microsoft.com>

dulikvor committed:
Microsoft.OperationalInsights: Property name change. (#11752)

* Microsoft.OperationalInsights: Proerty name change.

* Style fix.

Co-authored-by: Dudi Likvornik <dulikvor@microsoft.com>

msyyc committed:
[T2] devtestlabs for python track2 (#11724)

* python track2

* python

skayani committed:
Managed Services Swagger Consistency Fixes (#11471)

* Updated 2018-06-01-preview specs

* Updated 2019-04-01-preview specs

* Updated 2018-06-01-preview specs

* Updated 2020-02-01-preview specs

* Updated 2019-06-01 specs

* Updated 2019-09-01 specs

* Updated 2018-06-01-preview examples

* Updated ErrorResponse model

* Model validation fix

* Removed readOnly=true for ErrorResponse

* Updated type specification for error

* Removed old type reference for ErrorResponse

asmaskar committed:
Changes to avoid Breaking changes for dotnet (#11739)

sanchitkum committed:
Microsoft.Compute: Include CloudServices Resource Swagger under API Version 2020-10-01-preview (#11704)

Co-authored-by: Sanchit Kumar <Sanchit.Kumar@microsoft.com>

msyyc committed:
[T2] datalakestore for python track2 (#11722)

* python track2

* python.md

shschwar commented:
> ### Swagger Validation Report
> `️❌`BreakingChange: 2 Errors, 0 Warnings failed [[Detail](https://github.com/Azure/azure-rest-api-specs/pull/11682/checks?check_run_id=1423996409)]
> 
> * Compared Swaggers (Based on Oad [v0.8.1](https://www.npmjs.com/package/@azure/oad/v/0.8.1))* original: [preview/2019-01-01-preview/SecurityInsights.json ](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json)<---> new: [ preview/2019-01-01-preview/SecurityInsights.json ](https://github.com/Azure/azure-rest-api-specs/blob/ff514cb5d8972587cd8bdfb53ca4b248a5d1fce5/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json )
> 
> 
> Rule	Message
> `` `❌` `` [1038 - AddedPath](https://github.com/Azure/openapi-diff/tree/master/docs/rules/1038.md)	The new version is adding a path that was not found in the old version.
> New: [Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json#L2194:5](https://github.com/Azure/azure-rest-api-specs/blob/ff514cb5d8972587cd8bdfb53ca4b248a5d1fce5/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json#L2194:5)
> `` `❌` `` [1038 - AddedPath](https://github.com/Azure/openapi-diff/tree/master/docs/rules/1038.md)	The new version is adding a path that was not found in the old version.
> New: [Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json#L2248:5](https://github.com/Azure/azure-rest-api-specs/blob/ff514cb5d8972587cd8bdfb53ca4b248a5d1fce5/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json#L2248:5)
> `️❌`LintDiff: 1 Errors, 4 Warnings failed [[Detail](https://github.com/Azure/azure-rest-api-specs/pull/11682/checks?check_run_id=1423996431)]
> `️️✔️`Avocado succeeded [[Detail](https://github.com/Azure/azure-rest-api-specs/pull/11682/checks?check_run_id=1423996486)] [Expand]
> `️❌`ModelValidation: 5 Errors, 0 Warnings failed [[Detail](https://github.com/Azure/azure-rest-api-specs/pull/11682/checks?check_run_id=1423996523)]
> `️❌`SemanticValidation: 2 Errors, 0 Warnings failed [[Detail](https://github.com/Azure/azure-rest-api-specs/pull/11682/checks?check_run_id=1423996543)]
> `️️✔️`[Staging] Cross Version BreakingChange (Base on preview version) succeeded [[Detail](https://github.com/Azure/azure-rest-api-specs/pull/11682/checks?check_run_id=1423996576)] [Expand]
> `️️✔️`[Staging] Cross Version BreakingChange (Base on stable version) succeeded [[Detail](https://github.com/Azure/azure-rest-api-specs/pull/11682/checks?check_run_id=1423996615)] [Expand]
> Posted by Swagger Pipeline | [How to fix these errors?](https://aka.ms/ci-fix)

I'm not sure why adding new path considering breaking change. Can you explain?

Commit:
Fix gaps

Commit:
Fix gaps

Commit:
Fix types

Commit:
Fix types

Commit:
Update example file

Commit:
Update example file

Commit:
Define query response

Commit:
Define query response

Commit:
Extract EntityItem properties to properties

Commit:
Extract EntityItem properties to properties

Commit:
Additional properties

Commit:
Additional properties

Commit:
Fix get insights request example

Commit:
Fix get insights request example

Commit:
Prettify

Commit:
Prettify

Commit:
EntityQueryKind - Add enum

Commit:
EntityQueryKind - Add enum

Commit:
Add columnsDefinitions,  queriesDefinitions to query schema

Commit:
Add columnsDefinitions,  queriesDefinitions to query schema

roytan-microsoft committed:
[Hub Generated] Review request for Microsoft.MixedReality to add version preview/2019-02-28-preview (#11626)

* add isDataAction to listOperations spec

* check if type is wrong

* Type should be boolean

* add identity and kind to the spatial accounts def

* isDataAction should be an enum

* move isDataAction to its own enum

* remove kind from list

* refer to common types instead of defining inline

* update schema to reflect reality

* remove extraneous curly brace

* add curly brace at the correct location

* change nameAvailable to boolean

* add a format for integer type

* change example

* updated wrong file

* remove format per breaking change discussion

* fix prettier error

zhenglaizhang committed:
update pr assignment (#11750)

ArcturusZhang committed:
Fix some bad output folder in go SDK (#11781)

Commit:
Fix typo

Commit:
Fix typo

brpanask committed:
Fix s360 kpis for 2020-03-01. (#11593)

* Fix s360 kpis for 2020-03-01.

* Delete Caches_Delete.json.bak

Remove backup file that was accidentally pushed.

* Delete Caches_Flush.json.bak

Remove backup file that was accidentally pushed.

* Delete Caches_Start.json.bak

Remove backup file that was accidentally pushed.

* Delete Caches_Stop.json.bak

Remove backup file that was accidentally pushed.

* Delete Caches_UpgradeFirmware.json.bak

Remove backup file that was accidentally pushed.

* Delete StorageTargets_Delete.json.bak

Remove backup file that was accidentally pushed.

corquiri committed:
Add change directory to reservations spec 2020-10-01-preview (#11555)

* Add change directory to reservations spec 2020-10-01-preview

* refactoring and updating description

* prettier and spellcheck

* Revert "prettier and spellcheck"

This reverts commit b2a9735fe4791b6202083c5d4111a2f588b5b947.

* Revert "refactoring and updating description"

This reverts commit 9dfdd7878629eb83a33070712b302bc1fba94f1e.

* Revert "Add change directory to reservations spec 2020-10-01-preview"

This reverts commit 064f7a9f86a72620d123d69a13c23ef8a61c38b7.

* moving change directory to new api version

* fixing lint error

* update default tag

Co-authored-by: Sean Oh <hyo@microsoft.com>

jaysterp committed:
[ACR] New API version 2020-11-01-preview: AZ and On-Prem support (#11412)

tunafish88 committed:
Adding managedBy property that is returned by the API but not in swagger (#11788)

yenchunchang committed:
Updates advisor.Json to improve Documentation for Suppressions_Get operation (#11433)

Co-authored-by: Yen-Chun Chang <yenchunchang@microsoft.com>

dhaiman committed:
AssociatedWorkspaces and CapacityReservationProperties added to cluster (#11736)

* AssociatedWorkspaces and CapacityReservationProperties added to cluster. Cluser defualt error bug fixed

* change AssociatedWorkspaces to array

* fix validation errors

* fix validation errors

* missing token added

* read only properties deleted

* read only properties deleted

* LintDIff Errors Fixed

* LACluster added to WorkspaceSku

* update error description

* revert types.json changes

* add prettier

* change default error response to use V2 format

* use V2 types.json file

Co-authored-by: Arie Olshanezky <arolshan@microsoft.com>

ninpan-ms committed:
add validate config server(stable) (#10858)

* add validate config server(stable)

* prettier

* typo

poetw committed:
BRP prune backup store API (#11479)

* Add backup external store pruning to the API spec

* Fix CI

MiriBerezin committed:
Marketplace miberezi 2020 (#11787)

* FIxing Privatestore ,Offer properties and examples

* Fixing properties types

* Fixing camel cases

* Add Idempotence to custom-words.txt

* prettier check changes

* adding description to isGov

* resolve conflicts

* Fix updateSuppressedDueIdempotence type

* FIx swagger validation issues

* Fix json

* Fix indentation conflicts

* Fix corruption

* Fix validations

* FIx marketplace swagger

MichalHel committed:
Download files for cm sensors (#11635)

* Added API of DownloadQuotaData to iotDefenderSettings and DownloadResetPassword to iotSensors

* Added reset password api also for the on premise iot sensors

* renamed DownloadQuotaData to DownloadManagerActivation

* Fixed the body in the DownloadResetPassword API

guy-microsoft committed:
Azure Healthbot swagger (#11681)

* Azure Healthbot swagger .

* Fixes according to review.

* Fixes according to review.

* Fixes according to review.

JackTn committed:
Update samplereadme.md (#11800)

fix language readme path

lavinir committed:
Updating Package Manifest Schema in Private Preview API (#11688)

nivats committed:
tried fixing packge infomration flagging (#10870)

Commit:
[Hub Generated] Review request for Microsoft.Maps/DEM to add version preview/1.0 (#11410)

* Adds base for updating Microsoft.Maps from version Alias/preview to version 2020-12-01-preview

* Updates readme

* Updates API version in new specs and examples

* Add DEM API reference docs

* Fixing Typos & format

* Fix Typos

* Add dem in render.json

* apply comments

* apply comments

* Apply comments

* Fix validation

* Revert change about version

Co-authored-by: Yuqi Zhang <yuqzhan@microsoft.com>

rarayudu committed:
[Hub Generated] Review request for Microsoft.Logic to add version stable/2019-05-01 (#11627)

* Adding integration service environment encryption configuration details to Logic spec

* Correcting json format errors

* Correcting json formats

* Updating examples

* Fixing encryption key properties definition

Co-authored-by: Rama Krishna Rayudu <rarayudu@microsoft.com>

cataggar committed:
fix powerplatforrm ref & duplicates (#11580)

wwendyc committed:
Microsoft.ProviderHub GA API version (#11775)

* RPaaS 2020.11.20 GA specs

* update readme

* prettier fix and add custom words

* spelling

* update SDKs

* go sdk

Co-authored-by: Wendy Chang <wendychang@microsoft.com>

datr93 committed:
top level name can be 64 char long (#11692)

ruowan committed:
Update comment.yml (#11797)

ericshape committed:
Update Azure SQLDB 2020-08-01-preview swagger files (#10994)

* Update 2020-08-01 preview swagger files
and fix lint error

* rollback the sql common type file reference.

* Fix RESPONSE_SCHEMA_NOT_IN_SPEC validation errors in example files

* address part of example files errors.verify the fixes.

* fix LocationCapabilityListByLocation.json

* fix OBJECT_ADDITIONAL_PROPERTIES

* update with most of fix

* BackupShortTermRetentionPolicies fix

* public only

* fix server Key with internal PR

* change resource group name DatabaseVulnerabilityAssessmentScans

diyakobo committed:
Dev saas microsoft.saa s 2018 03 01 beta (#11680)

* from local

Signed-off-by: Dina Yakobovich <diyakobo@microsoft.com>

* add examples

* fix saasDelete

* fix again saasDelete

* change to lowerCase

* second change to lower Case

* fix parameters

* update deleteOptions definition

* add folder

* add new line and cSpell

* typing errors

* change example rsponse

* change response of application example

* add statuses to responses

* change put payload parameters name

* change tag properties

* add readme files

* move x-ms-exaples to the right place

* add defoult response

* fix error responses

* change operation schema

* add 200 to delete response

* fix worning

* readme files

* remove long running

* add space

* spelling errors

* comments fixing

* return operation

* readonly

* add suppression

audunn committed:
[NetAppFiles] Anf 8187 update swagger rest api to 2020 08 01 (#11689)

* Add 2020-08-01 folder

* Update api version

* update readme

* update readme

* issue in readme

* issue in readme

* issue in readme

* touch

Co-authored-by: Audunn Baldvinsson <audunn.baldvinsson@netapp.com>

diyakobo committed:
fix typo issue (#11818)

subbarayudukamma committed:
[Hub Generated] Review request for Microsoft.Maps/Microsoft.Maps to add version preview/1.0 (#11745)

* Adding string type suppport for feature state style rule

* Making the Integer, Boolean and string rule types appear in a strange way. I think this is better than not having them on the doc. Working with the team to figure out a way to do it better.

* Cleaning up the workarounds. Added a link to the reference page.

* Minor change to the text.

* Correctign the stateset description to say that a feature can have only one state set at a given point.

* Prettier changes.

* One more prettier change. By the tool.

Co-authored-by: Subbarayudu Kamma <skamma@microsoft.com>

ruowan committed:
Update pull_request_assignment.yml (#11814)

colawwj committed:
correct-package-name (#11666)

majastrz commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -2191,6 +2191,108 @@
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/entities/{entityId}/queries": {
+      "get": {
```
>get [](start = 7, length = 3)

How many of these queries do you expect to see in the response? Do you need paging support here? Also see [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi009-always-add-x-ms-pageable-to-list-calls](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi009-always-add-x-ms-pageable-to-list-calls).

majastrz replied:
There's also info about it at [https://github.com/Azure/azure-resource-manager-rpc/blob/eec9f32c0357d0421be13c82ced767943013901a/v1.0/resource-api-reference.md#get-resource](https://github.com/Azure/azure-resource-manager-rpc/blob/eec9f32c0357d0421be13c82ced767943013901a/v1.0/resource-api-reference.md#get-resource) (search for "nextLink")

---
In reply to: [530723118](https://github.com/Azure/azure-rest-api-specs/pull/11682#discussion_r530723118) [](ancestors = 530723118)

majastrz commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/examples/entities/GetQueries.json:
```diff
@@ -0,0 +1,457 @@
+      "body": {
+        "value": [
+          {
+            "id": "/subscriptions/d0cfe6b2-9ac0-4464-9919-dccaee2e48c0/resourceGroups/myRg/providers/Microsoft.OperationalInsights/workspaces/myWorkspace/providers/Microsoft.SecurityInsights/queries/6db7f5d1-f41e-46c2-b935-230b36a569e6",
```
>6db7f5d1-f41e-46c2-b935-230b36a569e6 [](start = 199, length = 36)

If you do a GET on this ID, will this fail? You don't seem to have a path that allows an individual GET on a query resource.

majastrz commented on specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2019-01-01-preview/SecurityInsights.json:
```diff
@@ -2191,6 +2191,108 @@
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{operationalInsightsResourceProvider}/workspaces/{workspaceName}/providers/Microsoft.SecurityInsights/entities/{entityId}/getInsights": {
+      "post": {
```
>post [](start = 7, length = 4)

Why is it modeled as a POST? Can it be modeleted as an `insights` nested resource type like `queries`?

dochung4 committed:
Make DedicatedHostGroup.properties.supportAutomaticPlacement default to false (#11697)

* Removed maximum for DedicatedHostGroup

* Default supportAutomaticPlacement to false

avinashakka committed:
Updating vmSize description to correct hyperlinks and add a note on deprecating VirtualMachineSizeTypes enum in subsequent api-versions. (#11595)

* Removing vmSizeTypes enum

* Updating description and reverting the enum removal which will be done in a new api-version as a breaking change.

Co-authored-by: Avinash Akka <avakka@microsoft.com>

msyyc committed:
python track2 (#11728)

jianyexi committed:
Update ci-fix.md (#11828)

sirireddy12 committed:
Aggregated swagger changes k8connect (#11817)

* aggregated swagger changes k8conect

* prettier fix

Co-authored-by: Siri Teja Reddy Kasireddy <sikasire@microsoft.com>

ArcturusZhang committed:
Fix errors in various of readme files encountered in the recent go SDK release (#11826)

* Fix an issue on windows

* Fix cdn

* Fix cognitiveservices/customVision

* Fix cognitiveservices/personalizer

* Fix cognitiveservices/translatorText

* Fix data plane digitaltwins

* Remove problematic tag from release - no impact on sdk, this has never been released

* amend Fix cognitiveservices/translatorText

* Fix imds

* Fix migrate readme that are mistakenly damaged in #10305

* add one api-version back

* Fix digtaltwins data plane again

* Fix hybridnetwork

* Fix streamanlytics

* Revert some changes of digitaltwins

* Revert imds

* Fix migrate

* Fix migrate again

zhenglaizhang committed:
remove pipeline old un-needed tasks (#11730)

ChenTanyi committed:
add privateLinkResources and privateEndpointConnection for 2020-09 API (#11837)

pratimaupadhyay02 committed:
Updating existing CRR APIs and protection APIs (#10962)

* Fixing bug in PrepareDataMoveResponse Class

* Updating existing CRR and protection APIs

* Adding examples

* Updating APIs to 2020-07-01

* Updating conflict

* Resolving failures

* Update

* Update

* Update

* Update

* Adding workloadCRRAccessToken

* Adding discriminator for CRRAccesToken

* Update

* Reverting change to 2020-07-01

* Update

* Update

* Fixing prettier

ArcturusZhang committed:
Fix attestation output folder (#11838)

pratimaupadhyay02 committed:
Combining existing versions in latest version 2020-10-01 (#11733)

* Combining existing versions in latest version 2020-10-01

* Correcting version

* Fixing prettier

* Adding 2018-12-20 in latest package

ayfathim committed:
ASR: removing blob uri and sas token from examples (#11834)

* removing blob uri and sas token from examplaes

* added place holder value

ArcturusZhang committed:
Remove api-version in recoveryservicebackup (#11846)

00Kai0 committed:
add deploymentmanager track2 config (#11845)

Tiano2017 committed:
[Resources] Add extended locations in the /providers/resourceTypes API in 2020-08-01 version (#10986)

* copy everything from 2020-06-01

* replace 2020-06-01 with 2020-08-01

* add the new API /resourceTypes. add "locationMappings" property.

* add examples

* update readme.md

* run prettier

* fix tag for package-templatespecs

* Add changes to API version 2020-10-01

ori-licht committed:
[EntityQueries] - Adding EntityQueryKind to EntityQueries and changing Properties to Expansion properties (#11776)

* update swagger - aading LogicAppResourceId to Action put and response

* update swagger - changing to logicAppResourceId style

* adding LogicAppResourceId to Actions Request & Response

* adding LogicAppResourceId to Actions Request & Response

* adding Action property base

* editing examples and adding reauired to logicAppResourceId

* from LogicAppResourceId to logicAppResourceId

* running prettier

* running prettier

* removing required for LogicAppRexourceId

* changing logicApp propery doc

* changing logicApp propery description

* adding kind to entity queries and changing entityQueriesProperties to ExpansionEntityQueriesProperties

* changin return for entityQueryById

sanjaiganesh committed:
added new rules (#11677)

changlong-liu commented:
/azp run

mbialecka committed:
Fix the titles of Azure Stack HCI examples (#11853)

ShirleyJ27 committed:
Adding AFDX swagger and its examples (#11655)

* Adding AFDX swagger and its examples

* fix examples and added odata back

* added readme and fixed small sematic problems

* fix log analytics

* updates

* updates

* updates

* updates

* updates

* updates

* updates

* UPDATES

* add secret version back

* lint errors 1

* lint error 2

* lint 3

* lint 4

* updates

* spell check

* add origingroupoverrideaction

* address comments

* fixes

* updates

* remove ruleset update

* updates

Co-authored-by: Shirley Jiang <xujia@microsoft.com>

ArcturusZhang committed:
Add some profile tag for go SDK to serve the new profile 2020-09-01 (#10923)

* some update of the tags for profiles

* final update for profiles

* a fix of tag name

* Remove the changes for web readme since it does not work

* Rename tag for compute, revert keyvault tag change

* update

jianyexi committed:
Adding lint rules: R4028,R4029,R4030 (#11812)

* add R4028

* update doc (#11843)

* update doc

* update description

* format

* add scope

Co-authored-by: jianye xi <jianyxi@microsoft.com>
Co-authored-by: Ruoxuan Wang <52271048+ruowan@users.noreply.github.com>

solankisamir committed:
update examples (#11807)

Commit:
adding throttle analytics export API two new filters  (#11806)

* updating with two new filters

* fix typo

* typo

Commit:
Merge branch 'extend_insights_api' of https://github.com/shschwar/azure-rest-api-specs into extend_insights_api
